src/plot.js

```diff
@@ -756,24 +760,8 @@ function applyScaleTransforms(channels, options) {
 function inferChannelScale(channels) {
   for (const name in channels) {
     const channel = channels[name];
-    let {scale} = channel;
-    if (scale === true) {
-      switch (name) {
-        case "fill":
-        case "stroke":
-          scale = "color";
-          break;
-        case "fillOpacity":
-        case "strokeOpacity":
-        case "opacity":
-          scale = "opacity";
-          break;
-        default:
-          scale = scaleRegistry.has(name) ? name : null;
-          break;
-      }
-      channel.scale = scale;
-    }
+    const {scale} = channel;
+    if (scale === true) channel.scale = knownChannels.get(name).scale;
```

Comment:
👏

Reply:
Though maybe we want to handle an unknown channel as before?

```suggestion
    if (scale === true) channel.scale = knownChannels.get(name)?.scale ?? null;
```